lib/chef/application/solo.rb

```diff
@@ -215,39 +229,44 @@ def reconfigure
 
     Chef::Config[:solo] = true
 
-    if Chef::Config[:daemonize]
-      Chef::Config[:interval] ||= 1800
-    end
+    Chef::Log.deprecation("-r MUST be changed to --recipe-url, the -r option will be changed in Chef 13.0") if ARGV.include?("-r")
 
-    Chef::Application.fatal!(unforked_interval_error_message) if !Chef::Config[:client_fork] && Chef::Config[:interval]
+    if !Chef::Config[:solo_legacy_mode]
+      Chef::Config[:local_mode] = true
+    else
 
```

Comment:
could everything in this half of the conditional go in method like `configure_legacy_solo` or whatever to better balance the conditional?